app/controllers/posts_controller.rb

```diff
@@ -8,4 +8,29 @@ def show
     @user = User.find(params[:user_id])
     @post = @user.posts.find(params[:id])
   end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    puts 'Current User'
+
+    @post = Post.new(post_params)
+    @post.author = current_user
+
+    respond_to do |_format|
+      if @post.save
+        redirect_to user_posts_path(current_user)
```

Comment:
- [ ] **[OPTIONAL]** From your code a post is created users should be redirected to the post, but that does not happen, the app remains on the new post page. You may want to look into that.

  ![image](https://github.com/iamsjunaid/quill-quest/assets/36542986/4a9a26b1-99c0-417d-895a-9a34928801b4)